src/tests/src/libs/antares/study/parameters/parameters-tests.cpp

```diff
@@ -0,0 +1,227 @@
+/*
+** Copyright 2007-2024, RTE (https://www.rte-france.com)
+** See AUTHORS.txt
+** SPDX-License-Identifier: MPL-2.0
+** This file is part of Antares-Simulator,
+** Adequacy and Performance assessment for interconnected energy networks.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** Mozilla Public Licence 2.0 for more details.
+**
+** You should have received a copy of the Mozilla Public Licence 2.0
+** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+*/
+#define BOOST_TEST_MODULE "test parameters"
+#define BOOST_TEST_DYN_LINK
+
+#include <boost/test/unit_test.hpp>
+#include <filesystem>
+#include <iostream>
+#include <fstream>
+
+#include <antares/study/study.h>
+
+using namespace Antares::Data;
+
+// =================
+// The fixture
+// =================
+struct Fixture
+{
+    Parameters p;
+    std::filesystem::path path;
+    StudyLoadOptions options;
+    StudyVersion version = StudyVersion::latest();
+
+    Fixture()
+    {
+        path = std::filesystem::temp_directory_path() / "generaldata.ini";
+    }
+
+    void writeValidFile();
+    void writeInvalidFile();
+};
+
+
+// ==================
+// Tests section
+// ==================
+
+BOOST_AUTO_TEST_SUITE(parameters_tests)
+
+BOOST_FIXTURE_TEST_CASE(reset, Fixture)
+{
+    p.reset();
+    BOOST_CHECK_EQUAL(p.simulationDays.first, 0);
+    BOOST_CHECK_EQUAL(p.nbTimeSeriesThermal, 1);
+    BOOST_CHECK_EQUAL(p.synthesis, true);
+    BOOST_CHECK_EQUAL(p.ortoolsSolver, "sirius");
+}
+
+BOOST_FIXTURE_TEST_CASE(loadValid, Fixture)
+{
+    options.ortoolsSolver = "xpress";
+
+    writeValidFile();
+    p.loadFromFile(path.string(), version, options);
+
+    BOOST_CHECK_EQUAL(p.nbYears, 5);
+    BOOST_CHECK_EQUAL(p.seed[seedTsGenThermal], 5489);
+    BOOST_CHECK_EQUAL(p.include.reserve.dayAhead, true);
+    BOOST_CHECK_EQUAL(p.ortoolsSolver, "xpress");
+}
+
+BOOST_FIXTURE_TEST_CASE(fixBadValue, Fixture)
+{
+    p.nbYears = 100;
+    p.derated = true;
+    p.nbTimeSeriesThermal = 0;
+
+    p.fixBadValues();
+
+    BOOST_CHECK_EQUAL(p.nbYears, 1);
+    BOOST_CHECK_EQUAL(p.nbTimeSeriesThermal, 1);
+
+    p.nbYears = 10000000;
+    p.derated = false;
+
+    p.fixBadValues();
+
+    BOOST_CHECK_EQUAL(p.nbYears, 100000);
+}
+
+BOOST_FIXTURE_TEST_CASE(invalidValues, Fixture)
+{
+    writeInvalidFile();
+    p.loadFromFile(path.string(), version, options);
```

Comment:
Test return value to see if an error is emitted when encountering an invalid file.